lib/TBDGen/TBDGen.cpp

```diff
@@ -137,14 +137,35 @@ void TBDGenVisitor::visitPatternBindingDecl(PatternBindingDecl *PBD) {
 }
 
 void TBDGenVisitor::visitValueDecl(ValueDecl *VD) {
-  if (isPrivateDecl(VD))
-    return;
-
   auto declRef = SILDeclRef(VD);
 
-  // Transparent symbols don't exist, even in public.
-  // FIXME: this should really be "is SIL only".
-  if (declRef.isTransparent())
+  bool isPrivate = isPrivateDecl(VD);
+  bool checkSILOnly = true;
+  // Even private methods of open classes (specifically, private methods that
+  // are in the vtable) have public symbols, because external subclasses
+  // currently need to refer to them by symbol for their own vtable.
+  switch (declRef.getSubclassScope()) {
+  case SubclassScope::External:
+    // Constructors retain their normal linkage behaviour.
+    if (isa<ConstructorDecl>(VD))
+      break;
+
+    // Unlike the "truly" public things, private things have public symbols
```

Comment:
Public transparent things are serialized, so they can always be called directly or re-emitted as necessary. But yeah, it's pretty goofy and we'd like to clean it up. cc @swiftix